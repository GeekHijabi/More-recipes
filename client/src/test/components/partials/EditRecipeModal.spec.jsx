import React from 'react';
import { shallow } from 'enzyme';
import { EditRecipeModal }
  from '../../../js/components/Partials/EditRecipeModal';

const recipeInput = {
  recipeName: 'rice',
  ingredients: 'rice, water',
  description: 'boil rice',
  imageUrl: '',
  errors: {},
  errorMessage: '',
  isLoading: false,
  hasError: false
};

const props = {
  editRecipe: jest.fn(() => Promise.resolve({ ok: true })),
  recipe: {},
  toggle: jest.fn(() => Promise.resolve({ ok: true })),
  isOpen: true,
  isLoadingRecipe: true
};
describe('Component: EditRecipeModal', () => {
  beforeEach(() => {
    global.toastr = {
      success: () => {},
      error: () => {}
    };
  });

  describe('EditRecipeModal snapshot', () => {
    it('should render correctly', () => {
      const signIn = shallow(<EditRecipeModal {...props} />);
      expect(signIn).toMatchSnapshot();
    });
  });

  describe('onChange', () => {
    it('should change event', () => {
      const event = {
        target: { name: 'ingredients', value: 'rice, water' },
      };
      const wrapper = shallow(<EditRecipeModal {...props} />);
      wrapper.instance().onChange(event);
      expect(wrapper.instance().state.ingredients).toBe('rice, water');
    });
  });

  describe('onSubmit', () => {
    it('should add recipe when recipe details is set to the state', () => {
      const event = {
        preventDefault: jest.fn()
      };
      const submit = shallow(<EditRecipeModal {...props} />);
      submit.setState(recipeInput);
      submit.setState({ hasError: false });
      submit.instance().onSubmit(event);
    });
  });
});
