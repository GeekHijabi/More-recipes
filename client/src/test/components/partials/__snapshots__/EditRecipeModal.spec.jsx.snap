// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: EditRecipeModal EditRecipeModal snapshot should render correctly 1`] = `
<Modal
  autoFocus={true}
  backdrop={true}
  backdropTransition={
    Object {
      "mountOnEnter": true,
      "timeout": 150,
    }
  }
  fade={true}
  isOpen={true}
  keyboard={true}
  modalTransition={
    Object {
      "timeout": 300,
    }
  }
  onClosed={[Function]}
  onOpened={[Function]}
  role="dialog"
  toggle={[MockFunction]}
  zIndex={1050}
>
  <ModalHeader
    closeAriaLabel="Close"
    tag="h4"
    toggle={[MockFunction]}
    wrapTag="div"
  >
    Edit recipe
  </ModalHeader>
  <ModalBody
    tag="div"
  >
    <Form
      tag="form"
    >
      <FormGroup
        row={true}
        tag="div"
      >
        <Label
          for="exampleEmail"
          sm={4}
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Recipe Name
        </Label>
        <Col
          sm={8}
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            id="exampleEmail"
            name="recipeName"
            onChange={[Function]}
            tag="p"
            type="text"
          />
        </Col>
      </FormGroup>
      <FormGroup
        row={true}
        tag="div"
      >
        <Label
          for="exampleEmail"
          sm={4}
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Ingredients
        </Label>
        <Col
          sm={4}
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <textarea
            className="styledTextarea"
            id="exampleEmail"
            name="ingredients"
            onChange={[Function]}
            type="text"
          />
        </Col>
      </FormGroup>
      <FormGroup
        row={true}
        tag="div"
      >
        <Label
          for="exampleEmail"
          sm={4}
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Description
        </Label>
        <Col
          sm={8}
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <textarea
            className="styledTextarea"
            id="exampleEmail"
            name="description"
            onChange={[Function]}
            type="text"
          />
        </Col>
      </FormGroup>
      <FormGroup
        row={true}
        tag="div"
      >
        <Label
          for="exampleFile"
          lg={4}
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Image
        </Label>
        <Col
          sm={8}
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <input
            accept="image/*"
            name="image"
            onChange={[Function]}
            type="file"
          />
          <img
            alt="sample"
            className="styledImage"
            src={Object {}}
          />
        </Col>
      </FormGroup>
      <FormGroup
        check={true}
        className="modal-button"
        row={true}
        tag="div"
      >
        <Col
          sm={
            Object {
              "offset": 5,
              "size": 10,
            }
          }
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Button
            className="submit-btn"
            color="secondary"
            onClick={[Function]}
            tag="button"
          >
            Edit
          </Button>
        </Col>
      </FormGroup>
    </Form>
  </ModalBody>
</Modal>
`;
